### PR TITLE
enable node logs collection for gke provider

### PR DESCRIPTION
**Pull request by saikat-royc**
**What type of PR is this?**

**What this PR does / why we need it**:
This patch enables logs collection for e2e tests deployed on GKE cluster.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
We have two issues here:
1.  When collecting logs, the log-dump.sh exits due to MASTER_NAME not set. The patch fixes that issue by skipping log collection for master.
```
KUBE_MASTER_IP: 
KUBE_MASTER: 
/tmp/gcp-pd-driver-tmp584652997/kubernetes/cluster/log-dump/log-dump.sh: line 344: MASTER_NAME: unbound variable
Sep 25 21:23:05.944: INFO: Error running cluster/log-dump/log-dump.sh: exit status 1
```
2.  When kubetest is run with gke provider, https://github.com/kubernetes/test-infra/blob/master/kubetest/gke.go#L240, 
      KUBERNETES_PROVIDER is unset, and KUBERNETES_CONFORMANCE_PROVIDER is set to 'gke'. We can use 'KUBERNETES_CONFORMANCE_PROVIDER' to add logic in log-dump.sh to collect logs for GKE nodes. Eventually, the true north would be not to unset KUBERNETES_PROVIDER, and create a cluster/gke/util.sh with the helper functions. But that being a larger change, I am re-using KUBERNETES_CONFORMANCE_PROVIDER to enable logging for GKE nodes.

**Does this PR introduce a user-facing change?**:

```release-note
Enable node logs collection for 'gke' provider
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by wojtek-t** [comment]
@jkaniuk @tosi3k - FYI

**Comment by saikat-royc** [comment]
@jkaniuk @tosi3k  gentle reminder for review comments

File: cluster/log-dump/log-dump.sh [hunk 1]
```diff
@@ -386,6 +394,12 @@ function dump_nodes() {
   elif [[ ! "${node_ssh_supported_providers}" =~ "${KUBERNETES_PROVIDER}" ]]; then
     echo "Node SSH not supported for ${KUBERNETES_PROVIDER}"
     return
+  elif [[ "${KUBERNETES_CONFORMANCE_PROVIDER:-}" == 'gke' ]] && [[ -n "${NODE_INSTANCE_GROUP[@]:-}" ]]; then
+    for group in "${NODE_INSTANCE_GROUP[@]}"; do
+      node_names+=($(gcloud compute instance-groups managed list-instances \
```
Comment (msau42) [inline cluster/log-dump/log-dump.sh:399]: does this work in a multi-zone cluster?

**Comment by tosi3k** [comment]
>When kubetest is run with gke provider, https://github.com/kubernetes/test-infra/blob/master/kubetest/gke.go#L240,
KUBERNETES_PROVIDER is unset, and KUBERNETES_CONFORMANCE_PROVIDER is set to 'gke'.

That's true but when we run log dumping for the `gke` provider in kubetest, we actually wrap the log-dump.sh script and set `KUBERNETES_PROVIDER` back to `gke` - see [kubetest/gke.go#425](https://github.com/kubernetes/test-infra/blob/cc960e1e71944f3442eefad24f195fa645ef5842/kubetest/gke.go#L425).
In fact, in my understanding, we shouldn't end up with attempting to dump logs from masters for the GKE provider altogether - in kubetest/gke.go we explicitly define `log_dump_custom_get_instances` function that prevents dumping from GKE masters over there: [log-dump.sh#332](https://github.com/kubernetes/kubernetes/blob/75250ebeee8724887454f2a1c84fdd32832da8c2/cluster/log-dump/log-dump.sh#L332).

>Eventually, the true north would be not to unset KUBERNETES_PROVIDER, and create a cluster/gke/util.sh with the helper functions.

Actually, we want to deprecate `kubernetes/kubernetes/cluster` directory in the long term (including the log dumping bash script). We're currently investigating ways to make the log dumping phase of tests for GKE deployers (and for GCE as well) not rely on things located inside `k/k` repository.

**Comment by tosi3k** [comment]
@saikat-royc could you further describe a scenario in which you ran into `log-dump.sh` attempting to dump logs from GKE master(s)?

**Comment by saikat-royc** [comment]
@tosi3k  We are running k8s e2e tests for [Persistent Disk CSI driver](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/12ea362bb378bb52fbbbb064407889719092ffea/test/k8s-integration/main.go#L591) on GKE. 
an example kubetest command which is initiated looks like this:
```
kubetest --test --ginkgo-parallel --check-version-skew=false --test_args=--ginkgo.focus=External.Storage --ginkgo.skip=\[Disruptive\]|\[Serial\] --storage.testdriver=/usr/local/google/home/saikatroyc/go/src/sigs.k8s.io/gcp-compute-persistent-disk-csi-driver/test/k8s-integration/config/test-config.yaml -report-dir=/tmp/e2egkelogs/sc-standard --up=false --down=false --provider=gke --gcp-network=default --check-version-skew=false --deployment=gke --gcp-node-image=cos --gcp-network=default --cluster=gcp-pd-csi-driver-test-cluster --gke-environment=prod --gcp-zone=us-central1-c --gcp-project=mattcary-saikatroyc-test
```

At the end of the test, the dump of logs seem to be initiated from here:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L1182
```
		Logf("Dumping logs locally to: %s", dir)
		cmd = exec.Command(path.Join(TestContext.RepoRoot, "cluster", "log-dump", "log-dump.sh"), dir)
```

In this case, at the time of calling log dump, KUBERNETES_PROVIDER is not set. Only KUBERNETES_CONFORMANCE_PROVIDER is set to 'gke'.

File: cluster/log-dump/log-dump.sh (hunk 1 above)
Comment (saikat-royc) [reply cluster/log-dump/log-dump.sh:399]: @msau42  Good question I will try it out. I have only tested with a single zone cluster.
I see [here](https://github.com/kubernetes/test-infra/blob/master/kubetest/gke.go#L57) and [here](https://github.com/kubernetes/test-infra/blob/master/kubetest/gke.go#L592) that 
gke-single-zone-node-instance-group is true by default. We may need to set it to false.

**Comment by tosi3k** [comment]
Thanks for providing the details.
> @tosi3k We are running k8s e2e tests for [Persistent Disk CSI driver](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/12ea362bb378bb52fbbbb064407889719092ffea/test/k8s-integration/main.go#L591) on GKE.
> an example kubetest command which is initiated looks like this:
> 
> ```
> kubetest --test --ginkgo-parallel --check-version-skew=false --test_args=--ginkgo.focus=External.Storage --ginkgo.skip=\[Disruptive\]|\[Serial\] --storage.testdriver=/usr/local/google/home/saikatroyc/go/src/sigs.k8s.io/gcp-compute-persistent-disk-csi-driver/test/k8s-integration/config/test-config.yaml -report-dir=/tmp/e2egkelogs/sc-standard --up=false --down=false --provider=gke --gcp-network=default --check-version-skew=false --deployment=gke --gcp-node-image=cos --gcp-network=default --cluster=gcp-pd-csi-driver-test-cluster --gke-environment=prod --gcp-zone=us-central1-c --gcp-project=mattcary-saikatroyc-test
> ```
> 
> At the end of the test, the dump of logs seem to be initiated from here:
> https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L1182
> 
> ```
> 		Logf("Dumping logs locally to: %s", dir)
> 		cmd = exec.Command(path.Join(TestContext.RepoRoot, "cluster", "log-dump", "log-dump.sh"), dir)
> ```
> 
> In this case, at the time of calling log dump, KUBERNETES_PROVIDER is not set. Only KUBERNETES_CONFORMANCE_PROVIDER is set to 'gke'.

What we'd like to prevent here is running the `log-dump.sh` from the `hack/ginkgo-e2e.sh` level here and use the log dumping mechanism implemented in kubetest (reason for that is mentioned here: https://github.com/kubernetes/test-infra/pull/3974). Providing `--dump` flag set to `/workspace/_artifacts` to kubetest invocations in [your](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/12ea362bb378bb52fbbbb064407889719092ffea/test/k8s-integration/main.go#L591) usecase should help.

**Comment by saikat-royc** [comment]
Using the --dump flag in kubetest worked for collecting GKE node logs. Closing this PR

**Pull request closed without merging**